ql/src/java/org/apache/hadoop/hive/ql/parse/repl/dump/io/FileOperations.java

```diff
@@ -165,4 +178,83 @@ private void validateSrcPathListExists() throws IOException, LoginException {
       throw new FileNotFoundException(FILE_NOT_FOUND.format(e.getMessage()));
     }
   }
+
+  /**
+   * This needs the root data directory to which the data needs to be exported to.
+   * The data export here is a list of files either in table/partition that are written to the _files
+   * in the exportRootDataDir provided.
+   */
+   void exportFilesAsList() throws SemanticException, IOException, LoginException {
+    if (dataPathList.isEmpty()) {
+      return;
+    }
+    Retry<Void> retryable = new Retry<Void>(IOException.class) {
+      @Override
+      public Void execute() throws Exception {
+        try (BufferedWriter writer = writer()) {
+          for (Path dataPath : dataPathList) {
+            writeFilesList(listFilesInDir(dataPath), writer, AcidUtils.getAcidSubDir(dataPath));
+          }
+        } catch (IOException e) {
+          if (e instanceof FileNotFoundException) {
```

Comment:
It will retry for FileNotFoundException also

Reply:
no, it won't, right?

Reply:
FileNotFoundException is an IOException and you have added retry for IOException.

Reply:
if (e instanceof FileNotFoundException) {
            logger.error("exporting data files in dir : " + dataPathList + " to " + exportRootDataDir + " failed");
            throw new FileNotFoundException(FILE_NOT_FOUND.format(e.getMessage()));
          }

Reply:
Shouldn't this suffice?